src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/VSMSBuildNuGetProjectSystem.cs

```diff
@@ -401,27 +402,26 @@ public virtual void AddReference(string referencePath)
 
                     if (reference != null)
                     {
-                        var path = GetReferencePath(reference);
+                        dteOriginalPath = GetReferencePath(reference);
 
                         // If path != fullPath, we need to set CopyLocal thru msbuild by setting Private
                         // to true.
                         // This happens if the assembly appears in any of the search paths that VS uses to
                         // locate assembly references.
                         // Most commonly, it happens if this assembly is in the GAC or in the output path.
                         // The path may be null or for some project system it can be "".
-                        resolvedToGac = (!string.IsNullOrWhiteSpace(path)
-                            && !StringComparer.OrdinalIgnoreCase.Equals(path, assemblyFullPath));
+                        resolvedToPackage = !string.IsNullOrWhiteSpace(dteOriginalPath) && IsSamePath(dteOriginalPath, assemblyFullPath);
```

Comment:
Isn't the result the same? The raw XML is emitted manually to fix up the problem. It sounds like you are describing a new scenario where NuGet validates the DTE project state and can fail.